packages/shell-api/src/decorators.ts

```diff
@@ -141,56 +145,79 @@ interface Signatures {
   [key: string]: TypeSignature;
 }
 const signaturesGlobalIdentifier = '@@@mdb.signatures@@@';
-if (!global[signaturesGlobalIdentifier]) {
-  global[signaturesGlobalIdentifier] = {};
+if (!(global as any)[signaturesGlobalIdentifier]) {
+  (global as any)[signaturesGlobalIdentifier] = {};
 }
 
-const signatures: Signatures = global[signaturesGlobalIdentifier];
+const signatures: Signatures = (global as any)[signaturesGlobalIdentifier];
 signatures.Document = { type: 'Document', attributes: {} };
 
-export const toIgnore = [asPrintable, 'constructor'];
+type ClassSignature = {
+  type: string;
+  hasAsyncChild: boolean;
+  returnsPromise: boolean;
+  attributes: {
+    [methodName: string]: {
+      type: 'function';
+      serverVersions: [ string, string ];
+      topologies: Topologies[];
+      returnType: ClassSignature;
+      returnsPromise: boolean;
+      platforms: ReplPlatform[];
+    }
+  };
+};
+
+type ClassHelp = {
+  help: string;
+  docs: string;
+  attr: { name: string; description: string }[];
+};
+
+export const toIgnore = ['constructor'];
 export function shellApiClassDefault(constructor: Function): void {
   const className = constructor.name;
   const classHelpKeyPrefix = `shell-api.classes.${className}.help`;
-  const classHelp = {
+  const classHelp: ClassHelp = {
     help: `${classHelpKeyPrefix}.description`,
     docs: `${classHelpKeyPrefix}.link`,
     attr: []
   };
-  const classSignature = {
+  const classSignature: ClassSignature = {
     type: className,
     hasAsyncChild: constructor.prototype.hasAsyncChild || false,
     returnsPromise: constructor.prototype.returnsPromise || false,
     attributes: {}
   };
 
-  const classAttributes = Object.keys(constructor.prototype);
+  const classAttributes = Object.getOwnPropertyNames(constructor.prototype);
```

Comment:
Fun fact: When you transpile from TS to ES6 or above, class properties are non-enumerable, but for ES5 they were, see https://github.com/microsoft/TypeScript/issues/15038.